cmd/etcd-mesos-scheduler/app.go

```diff
@@ -122,12 +124,16 @@ func main() {
 		}
 	}
 
+	if *advertiseAddress == "" {
+		*advertiseAddress = *address
```

Comment:
Yeah, that's true. I don't believe the semantics will be different, but then I haven't tested it, so I'll happily update below to not set `PublishedAddress` if `advertiseAddress` is empty/nil :)